app/src/main/java/com/babylon/wallet/android/presentation/settings/securitycenter/ledgerhardwarewallets/LedgerHardwareWalletsScreen.kt

```diff
@@ -88,6 +89,21 @@ fun LedgerHardwareWalletsScreen(
             confirmText = stringResource(id = R.string.ledgerHardwareDevices_linkConnectorAlert_continue)
         )
     }
+
+    LaunchedEffect(Unit) {
+        addLinkConnectorViewModel.oneOffEvent.collect { event ->
+            when (event) {
+                is AddLinkConnectorViewModel.Event.Close -> {
+                    if (event.isConnectionEstablished) {
+                        viewModel.disableAddLedgerButtonUntilConnectionIsEstablished()
```

Comment:
Yes, it seems so.

Reply:
Addressed this by removing the redundant connection observing (now we have `isConnectionEstablished` flag for this). Thanks!

Reply:
this method and the hasAtLeastOneConnection from ledger messenger shouldn't be removed. 
The method `disableAddLedgerButtonUntilConnectionIsEstablished` disables the add ledger button as long as the wallet is setting up a link connection.  
But the problem here is not the method or the name, but the removal of the `isAnyLinkedConnectorConnected` which relies on the `PeerdroiClient` and now we rely on `PeerdroidLink`. 
That means that the `LedgerHardwareWalletsScreen` doesn't observe for any **actual** connected link connector but waits until the add link connection flow is done!
And what if the `PeerdroiClient` can't establish the connection immediately? User can still click on the "Continue" button of the "Add ledger device" screen and the message dispatch would fail!

Reply:
@giannis-rdx I see. You're right, as adding a link for sure doesn't mean the same thing as connecting to a link. I've tested this and I think because the connection happened so fast right after adding a link I didn't notice any problem (like the button staying enabled while the connection is happening). I'm reverting these changes. Thank you!